app/src/main/java/dev/arkbuilders/arkmemo/ui/dialogs/FilePickerDialog.kt

```diff
@@ -15,21 +13,33 @@ import dev.arkbuilders.arkfilepicker.presentation.filepicker.ArkFilePickerFragme
 import dev.arkbuilders.arkfilepicker.presentation.filepicker.ArkFilePickerMode
 import dev.arkbuilders.arkmemo.BuildConfig
 import dev.arkbuilders.arkmemo.R
+import dev.arkbuilders.arkmemo.ui.activities.MainActivity
+
+class FilePickerDialog: ArkFilePickerFragment() {
+
+    override fun dismiss() {
+        super.dismiss()
+        val activity = (requireActivity() as MainActivity)
+        if (activity.memoPreferences.getPathString() == null) activity.finish()
+    }
```

Comment:
Assuming `FilePickerDialog` might probably be used in other activity in the future, rather than `MainActivity`, this casting will cause a crash.
What do you think about below?



```suggestion
        val activity = (activity as? MainActivity)
        if (activity?.memoPreferences?.getPathString() == null) activity?.finish()
    }
```